library/src/test/java/com/getstream/sdk/chat/rest/ClientTest.java

```diff
@@ -0,0 +1,278 @@
+
+/*
+ * Created by Anton Bevza on 2019-10-23.
+ */
+public class ClientTest {
+
+    private static String TEST_API_KEY = "testApiKey";
+    private static String TEST_CLIENT_ID = "testClientId";
+
+    private MockWebServer mockWebServer;
+
+    @Spy
+    private TestTokenProvider testTokenProvider = new TestTokenProvider();
+    private Client client;
+
+    @BeforeEach
+    void initTest() throws IOException {
+        MockitoAnnotations.initMocks(this);
+        WebSocketServiceProvider webSocketServiceProvider = Mockito.mock(WebSocketServiceProvider.class);
+        Mockito.doReturn(new TestWebSocketService(null, "", null))
+                .when(webSocketServiceProvider).provideWebSocketService(
+                Mockito.any(), Mockito.any(), Mockito.any()
+        );
+        mockWebServer = new MockWebServer();
+        mockWebServer.start();
+        TestApiClientOptions testApiClientOptions =
+                new TestApiClientOptions(mockWebServer.url("/").toString());
+        client = new Client(TEST_API_KEY,
+                new ApiServiceProvider(testApiClientOptions),
+                webSocketServiceProvider,
+                new UploadStorageProvider(testApiClientOptions),
+                null);
+    }
+
+    @Test
+    void updateChannelSuccessTest() throws IOException, InterruptedException {
+        simulateConnection();
+        mockWebServer.enqueue(new MockResponse().setResponseCode(200).setBody(
+                new MockResponseFileReader("channel_success.json").getContent()
+        ));
+        final Object syncObject = new Object();
+        ChannelCallback callback = new ChannelCallback() {
+            @Override
+            public void onSuccess(ChannelResponse response) {
+                assertNotNull(response.getChannel());
+                synchronized (syncObject) {
+                    syncObject.notify();
+                }
+            }
+
+            @Override
+            public void onError(String errMsg, int errCode) {
+
+            }
+        };
+        HashMap<String, Object> extraFields = new HashMap<>();
+        extraFields.put("name", "testName");
+        extraFields.put("title", "testTitle");
+        Channel channel = new Channel(client, "testType", "testId", extraFields);
+        Message message = new Message();
+        message.setText("test message");
+        client.updateChannel(channel, message, callback);
+        RecordedRequest recordedRequest = mockWebServer.takeRequest();
+        assertEquals("/channels/testType/testId?api_key="
+                + TEST_API_KEY + "&client_id=" + TEST_CLIENT_ID, recordedRequest.getPath());
+        String expectedBody = "{\"data\":{\"name\":\"testName\",\"title\":\"testTitle\"}," +
+                "\"message\":{\"attachments\":[],\"text\":\"test message\"}}";
+        assertEquals(expectedBody, recordedRequest.getBody().readUtf8());
+
+        synchronized (syncObject) {
+            syncObject.wait();
```

Comment:
Synchronisation in tests is bad. Default `OkHttp` executor needed to be replaced with current thread executor.